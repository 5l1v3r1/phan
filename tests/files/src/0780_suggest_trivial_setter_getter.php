<?php

class HasGetter780 {
    private $propName = 'default';

    public function getPropName() : string {
        return $this->propName;
    }

    public function setPropName(string $value) : void {
        $this->propName = $value;;
    }
}
function test() {
    $g = new HasGetter780();
    echo $g->propName;
    $g->propName = 'newValue';
}
