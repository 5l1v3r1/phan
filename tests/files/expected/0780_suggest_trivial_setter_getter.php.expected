%s:13 PhanNoopStringLiteral Unused result of a string literal "x" near this line
%s:16 PhanNoopNumericLiteral Unused result of a numeric literal 0 near this line
%s:16 PhanTypeMissingReturn Method \HasGetter780::badMethod2 is declared to return string but has no return value
%s:22 PhanAccessPropertyPrivate Cannot access private property \HasGetter780->propName defined at %s:4 (Did you mean \HasGetter780::getPropName())
%s:23 PhanAccessPropertyPrivate Cannot access private property \HasGetter780->propName defined at %s:4 (Did you mean \HasGetter780::setPropName())
